Mobile Alpha 0.6.0 (33)

cvarley100 opened the pull request.

cvarley100 committed:
Mobile: Fix bar colours

cvarley100 committed:
Mobile: Add new receive page UI

cvarley100 committed:
Mobile: Clear timeouts on unmount

cvarley100 committed:
Mobile: Adjust receive page animations

cvarley100 committed:
Merge branch 'develop' into chore/mobile-bugs

cvarley100 committed:
Mobile: Rename component

cvarley100 committed:
Mobile: Fix progress bar on fingerprint authentication

cvarley100 committed:
Mobile: Fix topbar spacing

cvarley100 committed:
Mobile: Fix topbar opacity when disabled

cvarley100 committed:
Mobile: Fix Android alert padding when modal is open

cvarley100 committed:
Shared: Localise strings

cvarley100 committed:
Mobile: Update transaction history modal buttons

laumair commented on src/mobile/src/ui/components/AnimatedComponent.js:
```diff
@@ -145,6 +145,10 @@ class AnimatedComponent extends Component {
                     break;
+                case 'slideInBottom':
+                case 'slideOutBottom':
+                    animatedStyle = merge({}, animatedStyle, { width, transform: [{ translateY: this.slideValue }] });
```
Since `animatedStyle` is defined as inside this function and as a new object a deep merge can be avoided here and we can just reassign to `animatedStyle`

laumair commented on src/mobile/src/ui/components/TopBar.js:
```diff
@@ -233,6 +232,22 @@ class TopBar extends Component {
+     * Returns padding styling dependent on device type
+     *
+     * @method getTopbarPadding
+     * @returns {object}
```
The return type should be of type `any` since there's also a possibility that this function could return an `undefined` value (In case of iPhoneX).

laumair commented on src/mobile/src/ui/views/wallet/Receive.js:
```diff
@@ -748,6 +804,7 @@ const mapStateToProps = (state) => ({
     usdPrice: state.marketData.usdPrice,
     conversionRate: state.settings.conversionRate,
     hadErrorGeneratingNewAddress: state.ui.hadErrorGeneratingNewAddress,
+    ui: state.ui,
```
Misses `propType` validation.

cvarley100 committed:
Mobile: Adjust topbar scrollable

Commit:
Node Quorum (#631)

* Implement quorum for wereAddressesSpentFrom

* Simply #findSyncedNodes implementation

* Add quorum support for getBalances IRI endpoint

* Minor updates

- Add quorum support for getTrytes IRI endpoint
- Minor clean up in quorum methods

* Add quorum support for findTransactions IRI endpoint

* Rename quorum methods for better readability

* Integrate quorum methods with extended api

* Remove findTransactions & getTrytes endpoints from quorum

* Refactor quorum implementation and do minor fixes

- Update JSDoc typos
- Simplify quorum implementation (Remove duplications)
- Add a timeout for network request to each node
- Fix issues in findSyncedNodes implementation
- Update error messages

* Add coverage
- Add coverage for #determineQuorumResult
- Add coverage for #fallbackToSafeResult
- Add coverage for #findSyncedNodes

* Add empty payload checks in quorum methods

* Wrap percentage calculation in parentheses for clarity

Co-Authored-By: laumair <aquadestructor@icloud.com>

* Use develop branch of iota.lib.js

* Enforce quorum (by default) on supported methods

* Fix tests failing because of quorum enforcement

* Add code documentation and rename parameters & variables for clarity

- Related discussion https://github.com/iotaledger/trinity-wallet/pull/631#discussion_r241307209

* [Security] Bump cryptiles from 3.1.2 to 3.1.4 (#829)

* [Security] Bump nokogiri from 1.8.4 to 1.9.1 in /src/mobile/android (#828)

* Make sure accounts are always iterated in correct order (by account index) (#824)

Object.keys(<object>) function does not always preseve the order, especially if the object key starts with a number. This causes an issue when Object.keys is used for iterating on account names. https://github.com/iotaledger/trinity-wallet/pull/715 adds account indexes to state to make sure the order of accounts is always intact. However, some components in desktop use Object.keys directly on accounts object, which leads to certain issues of incorrect references to accounts. This commit fixes this issue by replacing Object.keys implementation on accounts with getAccountNamesFromState selector that guarantees the accounts order.

Fixes https://github.com/iotaledger/trinity-wallet/issues/811

Note that the issues Object.keys create are not always noticeable. Steps to reproduce these issues are:

- Add account with name "M"
- Add another account with name "0"
- Notice account names order in sidebar (Instead of "0" being the second account, it becomes the first)
- Generate receive address from account "M" (Instead of generating receive address for account "M", it generates receive address for account "0")

* New Crowdin translations [ci skip] (#826)

* New translations translation.json (Czech)

* New translations translation.json (Czech)

* New translations translation.json (Japanese)

* New translations translation.json (Spanish)

* Mobile: Update findSyncedNodes

* Update quorum.js

* Improve parameters and variable names

* Include custom nodes in quorum nodes

* Reduce node request timeout for getNodeInfo api calls (in quorum)

cvarley100 committed:
Mobile: Minor cleanup

cvarley100 committed:
Merge branch 'mobile-0.6.0-alpha-quorum' of https://github.com/iotaledger/trinity-wallet into mobile-0.6.0-alpha-quorum

rajivshah3 committed:
Mobile: Bump build numbers for release v0.6.0 (33) (#808)

cvarley100 committed:
Merge branch 'develop' into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Merge branch 'mobile-0.6.0-alpha-quorum' of https://github.com/iotaledger/trinity-wallet into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Mobile: Bump build no

cvarley100 committed:
Mobile: Fix account name opacity during certain tasks

cvarley100 committed:
Mobile: Adjust chart animations

cvarley100 committed:
Mobile: Adjust chart timeframe order

rajivshah3 committed:
Mobile: Bump build numbers for release 0.6.0 (35) (#844)

cvarley100 committed:
Resolve conflicts

Commit:
Merge branch 'develop' into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Mobile: Fix View Addresses scrollable area

Commit:
Fix failing tests (#879)

cvarley100 committed:
Mobile: Fix issues with animations in settings (#881)

* Mobile: Fix settings animations issues

* Mobile: Fix settings animations issues

* Mobile: Fade out on logout

* Mobile: Fix login animation direction

* Mobile: Remove unused prop

* Mobile: Fix notification icon when modal is open

* Mobile: Fix retry text opacity during promotion

Commit:
Quorum refinements (#884)

* Remove unnecessary node sync checks

With quorum integration, some node sync checks are now unnecessary as quorum will ensure we always query synced nodes. This commit removes these checks.

* Use default timeout for getNodeInfo while checking node's health

Commit:
Disable quorum for getLatestInclusion (#900)

- Add node health checks for account sync utils because getLatestInclusion support for quorum is (temporarily) disabled.

rajivshah3 committed:
Merge branch 'develop' into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Mobile: Add request-specific timeouts to quorum

cvarley100 committed:
Mobile: Add request-specific timeouts to quorum (#903)

* Mobile: Add request-specific timeouts to quorum

* Mobile: Update GET_NODE_INFO_REQUEST_TIMEOUT in config

cvarley100 committed:
Merge branch 'mobile-0.6.0-alpha-quorum' of https://github.com/iotaledger/trinity-wallet into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Alpha 0.6.0 (36) Release (#904)

* Mobile: Add request-specific timeouts to quorum

* Mobile: bump build no

cvarley100 committed:
Merge branch 'mobile-0.6.0-alpha-quorum' of https://github.com/iotaledger/trinity-wallet into mobile-0.6.0-alpha-quorum

Commit:
Fix findSyncedNodes implementation (#915)

findSyncedNodes wasn't filtering unsynced nodes from selected nodes in one case. This commit fixes the issue.

cvarley100 committed:
Fix modal layout issues on Android and iPhone X (#916)

* Mobile: Add request-specific timeouts to quorum

* Mobile: Fix modal layout issues and refactor

* Update src/mobile/src/ui/components/ModalView.js

Co-Authored-By: cvarley100 <cvarley100@gmail.com>

cvarley100 committed:
Merge branch 'mobile-0.6.0-alpha-quorum' of https://github.com/iotaledger/trinity-wallet into mobile-0.6.0-alpha-quorum

cvarley100 committed:
Merge branch 'develop' into mobile-0.6.0-alpha-quorum

rajivshah3 committed:
Merge branch 'develop' into mobile-0.6.0-alpha-quorum
